### DISCUSSION TITLE
resizing vector graphics files

### DISCUSSION THREAD

**Discussion by dev-slash-null** [ImageMagick/ImageMagick#2716] [Help] [answered]
Can vector graphics be resized in ImageMagick?  Attempting to resize a line results in the bitmap containing the line being resized instead:
$ magick -size 10x10 xc:white -draw "path 'M 0,0 l 10,10'" -resize 1000% x:
Does ImageMagick support vector graphic resizing?

**Comment by snibgo** [comment, marked as answer]
> Does ImageMagick support vector graphic resizing?

No.

IM is a raster image processor. "-draw" will update the raster image(s) by changing pixels.

However, "-path" is also "remembered" so if your output is SVG, the path will be copied to the SVG. But it won't be scaled.

**Comment by dev-slash-null** [comment]
Ok, so to resize a vector graphic like the one in my example, one would have to save as a .svg, and then resize with a vector image processor?  Do you know of any CLIs that could be used to resize such a graphics object in a pipeline (to/from imagemagick)?  It looks like Inkscape could be used in this manner.  Thanks.

**Comment by snibgo** [comment]
Inkscape can be run in batch, but I don't know if that includes resizing.

I don't know your final goal. Instead of writing an IM command that creates SVG, you could directly create a minimal SVG that contains your desired scaling.
